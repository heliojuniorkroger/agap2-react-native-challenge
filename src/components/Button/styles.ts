import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  menuButton: {
    width: '100%',
    height: 40,
    borderRadius: 20,
    justifyContent: 'space-between',
    alignItems: 'center',
    flexDirection: 'row',
    backgroundColor: '#003566',
    paddingHorizontal: 15,
  },
  menuButtonLabel: {
    fontFamily: 'Nunito-Bold',
    color: '#fff',
    fontSize: 14,
  },
});
