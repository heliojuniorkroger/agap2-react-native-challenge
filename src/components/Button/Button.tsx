import React, { useCallback } from 'react';
import { Text, Pressable, StyleProp, ViewStyle } from 'react-native';
import Icon from 'react-native-vector-icons/Feather';
import { styles } from './styles';

export type ButtonProps = {
  onPress: () => void;
  title: string;
  iconName?: string;
  style?: StyleProp<ViewStyle>;
};

export const Button = ({ onPress, style, title, iconName }: ButtonProps) => {
  const renderIcon = useCallback(() => {
    if (iconName) {
      return <Icon name={iconName} color="#fff" size={20} />;
    }

    return null;
  }, [iconName]);

  return (
    <Pressable style={[styles.menuButton, style]} onPress={onPress}>
      <Text style={styles.menuButtonLabel}>{title}</Text>
      {renderIcon()}
    </Pressable>
  );
};
