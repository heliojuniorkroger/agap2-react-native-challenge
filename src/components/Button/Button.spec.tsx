import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { Button } from './Button';

const createDefaultProps = () => ({
  title: 'Button',
  onPress: jest.fn(),
  testID: 'button-test-id',
  iconName: '',
  iconTestID: '',
});

describe('testing Button component', () => {
  it('should match snapshot', () => {
    const props = createDefaultProps();
    const result = render(<Button {...props} />);
    expect(result).toMatchSnapshot();
  });

  it('should call onPress function on pressing the button', async () => {
    const props = createDefaultProps();
    const { getByTestId } = render(<Button {...props} />);
    const pressable = getByTestId(props.testID);
    await fireEvent.press(pressable);
    expect(props.onPress).toBeCalled();
  });

  it('should render an icon when prop iconName is set', () => {
    const props = createDefaultProps();
    props.iconName = 'x';
    props.iconTestID = 'icon-test-id';
    const { getByTestId } = render(<Button {...props} />);
    const icon = getByTestId(props.iconTestID);
    expect(icon).toBeDefined();
  });
});
