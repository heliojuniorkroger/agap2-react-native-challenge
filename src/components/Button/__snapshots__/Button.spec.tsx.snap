// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Button component should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#003566",
        "borderRadius": 20,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "space-between",
        "paddingHorizontal": 15,
        "width": "100%",
      },
      undefined,
    ]
  }
  testID="button-test-id"
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Nunito-Bold",
        "fontSize": 14,
      }
    }
  >
    Button
  </Text>
</View>
`;
